File: siq/xml/src/main/java/com/sick/siq/xml/model/Round.java
package com.sick.siq.xml.model;

import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlProperty;
import lombok.Data;

@Data
public class Round {
    @JacksonXmlProperty(localName = "name", isAttribute = true)
    private String name;
    @JacksonXmlProperty(localName = "type", isAttribute = true)
    private String type;
    private Themes themes;
}

File: core/src/main/kotlin/model/round.kt
package com.sick.model

enum class RoundType {
    Simple,
    Final,
}

data class Round(
    val name: String,
    val type: RoundType,
    val themes: List<Theme>,
)

data class Theme(
    val name: String,
    val questions: List<Question>,
)
